snuba/datasets/transactions.py

```diff
@@ -0,0 +1,87 @@
+from snuba.clickhouse import (
+    ColumnSet,
+    DateTime,
+    IPv4,
+    IPv6,
+    LowCardinality,
+    Materialized,
+    Nested,
+    Nullable,
+    String,
+    Tuple,
+    UInt,
+    UUID,
+    WithDefault,
+)
+from snuba.datasets import DataSet
+from snuba.datasets.schema import ReplacingMergeTreeSchema
+from snuba.datasets.transactions_processor import TransactionsMessageProcessor
+
+
+class TransactionsDataSet(DataSet):
+    def __init__(self):
+        columns = ColumnSet([
+            ('project_id', UInt(64)),
+            ('event_id', UUID()),
+            ('trace_id', UUID()),
+            ('span_id', UInt(64)),
+            ('transaction_name', String()),
+            ('transaction_hash', Materialized(
+                UInt(64),
+                'cityHash64(transaction_name)',
+            )),
+            ('transaction_op', LowCardinality(String())),
+            ('start_ts', DateTime()),
+            ('start_ms', UInt(16)),
+            ('finish_ts', DateTime()),
+            ('finish_ms', UInt(16)),
+            ('duration', Materialized(
+                UInt(32),
+                '((finish_ts - start_ts) * 1000) + (finish_ms - start_ms)',
+            )),
+            ('platform', LowCardinality(String())),
+            ('environment', Nullable(String())),
+            ('release', Nullable(String())),
+            ('dist', Nullable(String())),
+            ('ip_tup', Tuple([
+                ('v4', Nullable(IPv4())),
+                ('v6', Nullable(IPv6())),
+            ])),
+            ('user', WithDefault(
```

Comment:
There are many fields that are pretty much the same thing in the events dataset but implemented a little bit differently here - e.g. column types or naming convention might be a little bit different. Would it be worthwhile to try to align them at some point?

Reply:
Yes some are implemented differently. We would like them to be more like this dataset than events. So it is more likely events that may one day migrate towards this approach.

Reply:
@lynnagara: Ultimately, the original events schema was far too heavy handed, and we'd want to eventually align it, but there's not really a driving force behind it yet.